packages/protocol/contracts/stability/interfaces/IMentoExchange.sol

```diff
@@ -6,26 +6,19 @@ pragma solidity ^0.5.13;
  */
 interface IMentoExchange {
   /**
-   * @notice Returns the output amount and new bucket sizes for a given input amount.
-   * @param tokenIn Address of the token being used to pay for another one.
-   * @param tokenOut Address of the token being exchanged for.
+   * @notice Returns the output amount for a given input amount.
    * @param tokenInBucketSize Size of the tokenIn bucket.
    * @param tokenOutBucketSize Size of the tokenOut bucket.
    * @param amountIn Amount of tokenIn being paid in.
    * @return amountOut Amount of tokenOut that will be paid out.
-   * @return newtokenInBucketSize Size of the tokenIn bucket after the swap.
-   * @return newtokenOutBucketSize Size of the tokenOut bucket after the swap.
    */
   function getAmountOut(
```

Comment:
We should also have `getAmountIn`

Reply:
And we should remove `getUpdateBuckets`

Reply:
We also don't need to pass in the token addresses.

Reply:
And if you want to make this pass you can rename the existing interface into `IMentoExchangeLegacy` and use that in all existing implementations. And provide the new `IMentoExchange` as we need it. That way we can migrate to using the new interface in the respective implementations without turning the branch red.